Remove many unused bfloat16 functions in TH

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30422 Remove exp10 in TH (unused)
* **#30329 Remove many unused bfloat16 functions in TH**
* #30176 Migrate reciprocal from the TH to Aten (CUDA)

Differential Revision: [D18764281](https://our.internmc.facebook.com/intern/diff/D18764281)

xuhdev committed:
Remove many unused bfloat16 functions in TH

[ghstack-poisoned]

izdeby commented:
What is the need in removing these? All these will be needed in future once additional ops coverage will be added.

xuhdev committed:
Update on "Remove many unused bfloat16 functions in TH"




[ghstack-poisoned]

xuhdev commented:
What do you mean? Only bfloat16 versions of these functions exist.

xuhdev committed:
Update on "Remove many unused bfloat16 functions in TH"




[ghstack-poisoned]

xuhdev committed:
Update on "Remove many unused bfloat16 functions in TH"




[ghstack-poisoned]

xuhdev committed:
Update on "Remove many unused bfloat16 functions in TH"


Differential Revision: [D18764281](https://our.internmc.facebook.com/intern/diff/D18764281)

[ghstack-poisoned]

xuhdev committed:
Update on "Remove many unused bfloat16 functions in TH"


Differential Revision: [D18764281](https://our.internmc.facebook.com/intern/diff/D18764281)

[ghstack-poisoned]
